gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1508,6 +1509,19 @@ private Credential getCredential(Configuration config)
             .withPrefixes(CONFIG_KEY_PREFIXES)
             .get(config, config::get);
 
+    if (isNullOrEmpty(serviceAccountToImpersonate)) {
+      Map<String, String> serviceAccountMapping =
```

Comment:
Can we get this map from options object using `getImpersonationServiceAccounts()` method instead?